### PR TITLE
feat(react-components): Add Ghost more settings [BND3D-4592]

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change
Feat  

#### Jira ticket :blue_book:
https://cognitedata.atlassian.net/browse/BND3D-4592

## Description :pencil:

Add settings for Ghost mode in the settings menu. Also add the magic hook useGhostMode, to be used in seach.

## How has this been tested? :mag:

Run Storybook on achitecture
Open settings

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/base/concreteCommands/SettingsCommand.ts [hunk 1]
```diff
@@ -29,6 +31,9 @@ export class SettingsCommand extends BaseSettingsCommand {
     super();
 
     this.add(new SetQualityCommand());
+    // Cad models
+    this.add(new CadDividerCommand());
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/base/concreteCommands/SettingsCommand.ts:35]: Ghost mode isn't really CAD specific? :thinking:

File: react-components/src/architecture/base/concreteCommands/cad/SetGhostModeCommand.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { RenderTargetCommand } from '../../commands/RenderTargetCommand';
+import { type TranslateKey } from '../../utilities/TranslateKey';
+
+export class SetGhostModeCommand extends RenderTargetCommand {
+  // ==================================================
+  // OVERRIDES
+  // ==================================================
+
+  public override get tooltip(): TranslateKey {
+    return { key: 'GHOST_MODE', fallback: 'Ghost mode' };
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/base/concreteCommands/cad/SetGhostModeCommand.ts:14]: Do we remember to put this into translation?

File: react-components/src/architecture/base/renderTarget/CommandsController.ts [hunk 3]
```diff
@@ -133,7 +133,20 @@ export class CommandsController extends PointerEvents {
...
+    for (const command of this._commands) {
+      if (isInstanceOf(command, classType)) {
+        return command;
+      }
+      for (const descendant of command.getDescendants()) {
+        if (isInstanceOf(descendant, classType)) {
+          return descendant;
+        }
+      }
+    }
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/base/renderTarget/CommandsController.ts:149]: Shouldn't this method call itself?

File: react-components/src/hooks/useGhostMode.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  const command = renderTarget.commandsController.getCommandByTypeRecursive(SetGhostModeCommand);
+  useOnUpdate(command, () => {
+    if (command !== undefined) {
+      setChecked(command.isChecked);
+    }
+  });
+  if (command === undefined) {
+    return false;
+  }
+  return command.isChecked;
```
Comment (haakonflatval-cognite) [inline react-components/src/hooks/useGhostMode.tsx:24]: Strictly speaking, I think we should return `_isChecked` here, even though this probably also works

**Review by haakonflatval-cognite — CHANGES_REQUESTED** [review]
Just a few nitpicks

File: react-components/src/architecture/base/concreteCommands/cad/SetGhostModeCommand.ts (hunk 2 above)
Comment (nilscognite) [reply react-components/src/architecture/base/concreteCommands/cad/SetGhostModeCommand.ts:14]: Yes

File: react-components/src/architecture/base/concreteCommands/SettingsCommand.ts (hunk 1 above)
Comment (nilscognite) [reply react-components/src/architecture/base/concreteCommands/SettingsCommand.ts:35]: Fixed

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

**Pull request merged**